process/block/metablock.go

```diff
@@ -1626,8 +1641,11 @@ func (mp *metaProcessor) computeExistingAndRequestMissingShardHeaders(metaBlock
 	mp.hdrsForCurrBlock.mutHdrsForBlock.Lock()
 	defer mp.hdrsForCurrBlock.mutHdrsForBlock.Unlock()
 
-	for i := 0; i < len(metaBlock.ShardInfo); i++ {
-		shardData := metaBlock.ShardInfo[i]
+	for _, shardData := range metaBlock.ShardInfo {
+		if shardData.Nonce == 0 {
+			continue
```

Comment:
Do we not need to execute lines 1163-1670 before continue? Maybe later we need to read from mp.hdrsForCurrBlock.hdrHashAndInfo the genesis shard header(to save it or other stuff) , or we should add this skip everywhere.

Reply:
putting to hdrsForCurrBlock is done at checkShardHeadersValidity